sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -3514,9 +3556,12 @@ pi_result cuda_piextUSMEnqueueMemcpy(pi_queue queue, pi_bool blocking,
                                      pi_uint32 num_events_in_waitlist,
                                      const pi_event *events_waitlist,
                                      pi_event *event) {
+  // enforce that the queue is valid
   assert(queue != nullptr);
-  assert(dst_ptr != nullptr);
-  assert(src_ptr != nullptr);
+  // check that the source and destination pointers are valid
+  if (dst_ptr == nullptr || src_ptr == nullptr) {
```

Comment:
Please, return assert.

Reply:
This seems needed to pass the test that expects a throw: https://github.com/intel/llvm/blob/sycl/sycl/test/usm/memcpy.cpp#L53